### PR TITLE
fix(client) Fix bugs in the CLI

### PR DESCRIPTION
**Pull request by samwillis**
Fixes:
- `generate --with-migrations` now runs in a ephemeral container on random ports and random container name
- #825 Correctly handles ssl for db urls
- Allow the pg proxy to be on an alternative host to the sync service (required for the proxy tunnel)
- configurable container name
- correctly handles `PG_PROXY_PORT` with `http` or `http:123` value

### CODE REVIEW

File: clients/typescript/src/cli/docker-commands/command-psql.ts [hunk 1]
```diff
@@ -20,23 +20,24 @@ export function makePsqlCommand() {
   return command
 }
 
-export function psql(opts: GetConfigOptionsForGroup<'proxy'>) {
+export function psql(
+  opts: GetConfigOptionsForGroup<'proxy'> & GetConfigOptionsForGroup<'electric'>
```
Comment (kevin-dp) [inline clients/typescript/src/cli/docker-commands/command-psql.ts:24]: Would be nice if we could write `GetConfigOptionsForGroup<'proxy' | 'electric'>`

File: clients/typescript/src/cli/docker-commands/command-psql.ts [hunk 2]
```diff
@@ -20,23 +20,24 @@ export function makePsqlCommand() {
...
+) {
```
Comment (kevin-dp) [inline clients/typescript/src/cli/docker-commands/command-psql.ts:25]: Is the TODO on the line below still relevant?

**Review by kevin-dp — APPROVED** [review]
Looks all good to me 👍

File: clients/typescript/src/cli/docker-commands/command-psql.ts (hunk 1 above)
Comment (samwillis) [reply clients/typescript/src/cli/docker-commands/command-psql.ts]: Good spot, changed

File: clients/typescript/src/cli/docker-commands/command-psql.ts (hunk 2 above)
Comment (samwillis) [reply clients/typescript/src/cli/docker-commands/command-psql.ts:25]: Hmm, not really. I don't think we need to make the `electric-sql psql` work when the pg isn't inside an electric-sql managed container, far to much complexity. Removed the TODO.

**Pull request merged**